doc/api/net.md

```diff
@@ -2,74 +2,120 @@
+The `net.Server` class is used to implement TCP or local servers.
 
-`net.Server` is an [`EventEmitter`][] with the following events:
+All `net.Server` objects are [`EventEmitter`][] instances.
 
 ### Event: 'close'
 <!-- YAML
 added: v0.5.0
 -->
 
-Emitted when the server closes. Note that if connections exist, this
-event is not emitted until all connections are ended.
+The `'close'` event is emitted when the server closes. If active connections
+exist, this event is not emitted until all such connections are ended.
+
+The listener callback is called without passing any arguments.
+
+```js
+server.on('close', () => {
+  console.log('The server has been closed.');
+});
+```
 
 ### Event: 'connection'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {net.Socket} The connection object
+The `'connection'` event is emitted when a new connection is established.
 
-Emitted when a new connection is made. `socket` is an instance of
-`net.Socket`.
+The listener callback is called with the `net.Socket` object for the connection
+passed as the only argument.
+
+```js
+server.on('connection', (socket) => {
+  console.log('A connection has been established');
+});
+```
 
 ### Event: 'error'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {Error}
+The `'error'` event is emitted when an error occurs. The [`'close'`][] event
+will be emitted immediately following this event.
+
+The listener callback is called with the `Error` object passed as the only
+argument.
 
-Emitted when an error occurs.  The [`'close'`][] event will be called directly
-following this event.  See example in discussion of `server.listen`.
+```js
+server.on('error', (error) => {
+  console.log('An error occurred!', error);
+});
+```
 
 ### Event: 'listening'
 <!-- YAML
 added: v0.1.90
 -->
 
-Emitted when the server has been bound after calling `server.listen`.
+The `'listening'` event is emitted when the server has been bound after calling
+`server.listen`.
+
+The listener callback is called without passing any arguments:
+
+```js
+server.on('listening', () => {
+  console.log('The server is now listening for connections!');
+});
+```
 
 ### server.address()
 <!-- YAML
 added: v0.1.90
 -->
 
-Returns the bound address, the address family name, and port of the server
-as reported by the operating system.
-Useful to find which port was assigned when getting an OS-assigned address.
-Returns an object with `port`, `family`, and `address` properties:
-`{ port: 12346, family: 'IPv4', address: '127.0.0.1' }`
+The `server.address()` method returns the address to which the server is
+currently bound, the address family name, and port of the server as reported by
+the operating system. This is primarily useful as a means of determining which
+port was assigned when getting an OS-assigned address.
 
-Example:
+The method returns an object with the following properties:
+
+* `port` {number}
+* `family` {String} Will be either `'IPv4'` or `'IPv6'`
+* `address` {String} The IPv4 or IPv6 network address
+
```

Comment:
I know this wasn’t documented before, but it may be worth noting that `.address()` will return the path of a UNIX socket if that’s the type of the server.